cmd/server/api/rpchandler_test.go

```diff
@@ -0,0 +1,134 @@
+package api_test
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/freeverseio/laos-universal-node/cmd/server/api"
+	"github.com/freeverseio/laos-universal-node/cmd/server/api/mock"
+	"go.uber.org/mock/gomock"
+)
+
+func TestPostRpcHandler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockHttpClient := mock.NewMockHttpClientInterface(ctrl)
+	handler := api.NewApiHandler(
+		"https://polygon-mumbai.g.alchemy.com/",
+		api.WithHttpClient(mockHttpClient),
+	)
+
+	tests := []struct {
+		name           string
+		requestBody    string
+		mockResponse   string
+		responseGzip   bool
+		mockError      error
+		expectedStatus int
+		expectedBody   string
+	}{
+		{
+			name:           "successful request",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+			mockResponse:   `{"jsonrpc":"2.0","result":"1001","id":67}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","result":"1001","id":67}`,
+		},
+		{
+			name: "successful eth_call request with parrams",
+			requestBody: `{
+        "jsonrpc": "2.0",
+        "method": "eth_call",
+        "params": [{
+            "to": "0xc4d9faef49ec1e604a76ee78bc992abadaa29527",
+            "data": "0xc87b56dd0000000000000000000000000000000000000000000000000000000000000000"
+        }, "latest"],
+        "id": 1
+    }`,
+			mockResponse:   `{"jsonrpc":"2.0","id":1,"result":"0x00477777730000000000"}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","id":1,"result":"0x00477777730000000000"}`,
+		},
+		{
+			name:           "non successful request",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+			mockResponse:   `{"jsonrpc":"2.0","error":{"code":-32601,"message":"The method net_version does not exist/is not available"}}`,
+			expectedStatus: http.StatusOK,
+			expectedBody:   `{"jsonrpc":"2.0","error":{"code":-32601,"message":"The method net_version does not exist/is not available"}}`,
+		},
+		{
+			name:           "non successful request with invalid json",
+			requestBody:    `{"jsonrpc":"2.0","method":"net_version","params":[],"id":67`,
+			mockResponse:   `{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"}}`,
+			expectedStatus: http.StatusOK,
```

Comment:
this should return http.StatusOK even if error happens? I am just asking:)

Reply:
yes